### PR TITLE
🌱 Update Kustomize deprecated syntax

### PR DESCRIPTION
**Pull request by peppi-lotta**
Update Kustomize deprecated syntax:

- patchesStrategicMerge -> patches
- patchesJson6902 -> patches
- vars and varReference -> replacements
- bases -> resources

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by kashifest** [comment]
cc @lentzi90 @Rozzii @mboukhalfa @smoshiur1237 @adilGhaffarDev @kashifest

File: config/certmanager/certificate.yaml [hunk 1]
```diff
@@ -21,4 +21,4 @@ spec:
   issuerRef:
     kind: Issuer
     name: selfsigned-issuer
-  secretName: $(SERVICE_NAME)-cert # this secret will not be prefixed, since it's not managed by kustomize
+  secretName: capm3-webhook-service-cert # this secret will not be prefixed, since it's not managed by kustomize
```
Comment (smoshiur1237) [inline config/certmanager/certificate.yaml:24]: is there any reason to hard code it?

Comment (peppi-lotta) [reply config/certmanager/certificate.yaml:24]: @lentzi90 had a comment on this other PR about this: https://github.com/kubernetes-sigs/cluster-api/pull/9223#discussion_r1297093508

Basically replacements doesn't work exactly the same way as vars and dynamic values don't work out of the box.

**Review by lentzi90 — COMMENTED** [review]
/approve

File: config/crd/patches/cainjection_in_metal3machines.yaml [hunk 2]
```diff
@@ -4,5 +4,5 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    cert-manager.io/inject-ca-from: $(CERTIFICATE_NAMESPACE)/$(CERTIFICATE_NAME)
+    cert-manager.io/inject-ca-from: CERTIFICATE_NAMESPACE)/CERTIFICATE_NAME
```
Comment (kashifest) [inline config/crd/patches/cainjection_in_metal3machines.yaml:7]: same as before

**Review by kashifest — CHANGES_REQUESTED** [review]

**Review by Rozzii — APPROVED** [review]
/lgtm

**Comment by tuminoid** [comment]
/test metal3-centos-e2e-integration-test-main metal3-ubuntu-e2e-integration-test-main

/hold
To fix merge pool

**Comment by tuminoid** [comment]
/hold
Requested changes from @kashifest blocks merge pool.

**Review by kashifest — APPROVED** [review]

**Comment by Rozzii** [comment]
Looks like it has passed every test I will remove the hold
/hold cancel

**Pull request merged**